#! /bin/sh /usr/share/dpatch/dpatch-run
## 808_rexml_document_transitive.dpatch by akira yamada <akira@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: REXML::Document#write(io, 1, true, true) raises NameError/ArgumentError

@DPATCH@
diff -urNad ruby1.8-1.8.7.22~/source/lib/rexml/document.rb ruby1.8-1.8.7.22/source/lib/rexml/document.rb
--- ruby1.8-1.8.7.22~/source/lib/rexml/document.rb	2008-06-06 17:05:24.000000000 +0900
+++ ruby1.8-1.8.7.22/source/lib/rexml/document.rb	2008-09-09 22:06:37.000000000 +0900
@@ -185,6 +185,7 @@
       end
       formatter = if indent > -1
           if trans
+            require "rexml/formatters/transitive"
             REXML::Formatters::Transitive.new( indent, ie_hack )
           else
             REXML::Formatters::Pretty.new( indent, ie_hack )
diff -urNad ruby1.8-1.8.7.22~/source/lib/rexml/element.rb ruby1.8-1.8.7.22/source/lib/rexml/element.rb
--- ruby1.8-1.8.7.22~/source/lib/rexml/element.rb	2008-06-06 17:05:24.000000000 +0900
+++ ruby1.8-1.8.7.22/source/lib/rexml/element.rb	2008-09-09 22:06:57.000000000 +0900
@@ -675,6 +675,7 @@
       Kernel.warn("#{self.class.name}.write is deprecated.  See REXML::Formatters")
       formatter = if indent > -1
           if transitive
+            require "rexml/formatters/transitive"
             REXML::Formatters::Transitive.new( indent, ie_hack )
           else
             REXML::Formatters::Pretty.new( indent, ie_hack )
diff -urNad ruby1.8-1.8.7.22~/source/lib/rexml/formatters/transitive.rb ruby1.8-1.8.7.22/source/lib/rexml/formatters/transitive.rb
--- ruby1.8-1.8.7.22~/source/lib/rexml/formatters/transitive.rb	2007-07-30 11:26:17.000000000 +0900
+++ ruby1.8-1.8.7.22/source/lib/rexml/formatters/transitive.rb	2008-09-09 22:07:33.000000000 +0900
@@ -12,9 +12,10 @@
     # formatted.  Since this formatter does not alter whitespace nodes, the
     # results of formatting already formatted XML will be odd.
     class Transitive < Default
-      def initialize( indentation=2 )
+      def initialize( indentation=2, ie_hack=false )
         @indentation = indentation
         @level = 0
+        @ie_hack = ie_hack
       end
 
       protected
@@ -29,6 +30,7 @@
         output << "\n"
         output << ' '*@level
         if node.children.empty?
+          output << " " if @ie_hack
           output << "/" 
         else
           output << ">"
